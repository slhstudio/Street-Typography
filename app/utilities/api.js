import axios from 'axios';

const api = {}

api.isSignedIn = async (error) => {
  const result = await axios.get('/user')
    .catch(error);
  return result.data;
}

module.exports = api;