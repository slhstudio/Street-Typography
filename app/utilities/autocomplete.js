function autocomplete(input, latInput, lngInput) {
  if (!input) return;
  console.log(input);
  const dropdown = new google.maps.places.Autocomplete(input);
}

export default autocomplete;