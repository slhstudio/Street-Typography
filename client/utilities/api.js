import axios from 'axios';

export async function isSignedIn (error) {
  const result = await axios.get('/user')
    .catch(error);
  return result.data;
}

export async function findAllPhotos (error) {
  const result = await axios.get('/findAllPhotos')
    .catch(error);
  return result;
}

