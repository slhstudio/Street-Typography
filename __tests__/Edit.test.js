import React from 'react';
import EditComponent from '../client/components/Edit';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import { uploadChanges as mockUploadChanges, deletePhoto as mockDeletePhoto } from '../client/utilities/api';


const image = 'photo1.png';
const note = 'nice typeface';

const wait = (amount = 0) => new Promise(resolve => setTimeout(resolve, amount));

jest.mock('../client/utilities/api', () => {
  return {
    uploadChanges: jest.fn(() => Promise.resolve('updated')),
    deletePhoto: jest.fn(() => Promise.resolve('success'))
  }
})

describe('<Edit/>', () => {
  it ('matches the snapshot', () => {
    const wrapper = shallow(<EditComponent photo={image} info={note}/>);
    expect(toJSON(wrapper)).toMatchSnapshot();
    // const deleteButton = wrapper.find('.row').first().childAt(1).childAt(0);//
    // expect((deleteButton).type()).toBe('button');
    // expect((deleteButton).text()).toBe('DELETE');
  });
  it ('conditional render of editing form vs. redirect works properly', () => {
    const wrapper = shallow(<EditComponent photo={image} info={note}/>);
    wrapper.setState({ deleted : true });
    expect(toJSON(wrapper)).toMatchSnapshot();
    //expect(wrapper.find('Redirect').exists()).toBe(true);
    wrapper.setState({ deleted : false });
    expect(toJSON(wrapper)).toMatchSnapshot();
    //expect(wrapper.find('.row').exists()).toBe(true);
  });
  it ('conditional render of edit vs. save works properly', () => {
    const wrapper = shallow(<EditComponent photo={image} info={note}/>);
    wrapper.setState({ editing : false });
    expect(toJSON(wrapper)).toMatchSnapshot();
    // expect(wrapper.find('p').text()).toBe('nice typeface');
    // const changingButton = wrapper.find('.row').at(1).find('button');
    // expect((changingButton).text()).toBe('EDIT');
    wrapper.setState({ editing : true });
    expect(toJSON(wrapper)).toMatchSnapshot();
    // expect(wrapper.find('input').exists()).toBe(true);
    // const changedButton = wrapper.find('.row').at(1).find('button');
    // expect(changedButton.text()).toBe('SAVE');
  });
  it ('shows controlled input value correctly', () => {
    const wrapper = shallow(<EditComponent photo={image} info='weird lettering'/>);
    wrapper.setState({ editing : true });
    expect(toJSON(wrapper)).toMatchSnapshot();
    //expect(wrapper.find('input').props().value).toBe('nice typeface');
  });
  it ('changes state on click', () => {
    const wrapper = shallow(<EditComponent photo={image} info={note}/>);
    wrapper.find('.test1').simulate('click');
    expect(wrapper.state().editing).toBe(true);
  })
  it ('when editing, changes state on click and calls upload function', async () => {
    const wrapper = shallow(<EditComponent photo={image} info={note}/>);
    wrapper.setState({ editing : true });
    wrapper.find('.test2').simulate('click');
    //const result = await mockUploadChanges();
    await wait();
    expect(mockUploadChanges).toHaveBeenCalledTimes(1);
    expect(mockUploadChanges).toHaveBeenCalledWith(note);
    expect(wrapper.state().editing).toBe(false);
  });
  it ('when deleting, changes state on click and calls delete function', async () => {
    const wrapper = shallow(<EditComponent photo={image} info={note}/>);
    wrapper.find('.test3').simulate('click');
    await wait();
    expect(mockDeletePhoto).toHaveBeenCalledTimes(1);
    expect(mockDeletePhoto).toHaveBeenCalledWith(image);
    expect(wrapper.state().deleted).toBe(true);
  })

});