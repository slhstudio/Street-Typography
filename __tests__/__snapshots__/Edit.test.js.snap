// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Edit/> conditional render of edit vs. save works properly 1`] = `
<div>
  <div
    className="row"
  >
    <div
      className="row"
    >
      <p>
        nice typeface
      </p>
      <button
        className="test1"
        onClick={[Function]}
      >
        EDIT
      </button>
    </div>
    <div>
      <button
        className="test3"
        onClick={[Function]}
      >
        DELETE
      </button>
    </div>
  </div>
</div>
`;

exports[`<Edit/> conditional render of edit vs. save works properly 2`] = `
<div>
  <div
    className="row"
  >
    <div
      className="row"
    >
      <input
        onChange={[Function]}
        type="text"
        value="nice typeface"
      />
      <button
        className="test2"
        onClick={[Function]}
      >
        SAVE
      </button>
    </div>
    <div>
      <button
        className="test3"
        onClick={[Function]}
      >
        DELETE
      </button>
    </div>
  </div>
</div>
`;

exports[`<Edit/> conditional render of editing form vs. redirect works properly 1`] = `
<div>
  <Redirect
    push={false}
    to="/mine"
  />
</div>
`;

exports[`<Edit/> conditional render of editing form vs. redirect works properly 2`] = `
<div>
  <div
    className="row"
  >
    <div
      className="row"
    >
      <p>
        nice typeface
      </p>
      <button
        className="test1"
        onClick={[Function]}
      >
        EDIT
      </button>
    </div>
    <div>
      <button
        className="test3"
        onClick={[Function]}
      >
        DELETE
      </button>
    </div>
  </div>
</div>
`;

exports[`<Edit/> matches the snapshot 1`] = `
<div>
  <div
    className="row"
  >
    <div
      className="row"
    >
      <p>
        nice typeface
      </p>
      <button
        className="test1"
        onClick={[Function]}
      >
        EDIT
      </button>
    </div>
    <div>
      <button
        className="test3"
        onClick={[Function]}
      >
        DELETE
      </button>
    </div>
  </div>
</div>
`;

exports[`<Edit/> shows controlled input value correctly 1`] = `
<div>
  <div
    className="row"
  >
    <div
      className="row"
    >
      <input
        onChange={[Function]}
        type="text"
        value="weird lettering"
      />
      <button
        className="test2"
        onClick={[Function]}
      >
        SAVE
      </button>
    </div>
    <div>
      <button
        className="test3"
        onClick={[Function]}
      >
        DELETE
      </button>
    </div>
  </div>
</div>
`;
